crates/nostr-mls-storage/src/messages/parser.rs

```diff
@@ -0,0 +1,496 @@
+        let test_cases = vec![
+            (
+                "\t\t",
+                vec![
+                    SerializableToken::Text("\t\t".to_string()), // TODO: This should be updated upstream to handle tabs as whitespace
+                ],
+            ),
+            (
+                "  \t  ",
+                vec![
+                    SerializableToken::Whitespace,
+                    SerializableToken::Whitespace,
+                    SerializableToken::Text("\t ".to_string()), // TODO: This should be updated upstream to handle tabs as whitespace
+                    SerializableToken::Whitespace,
+                ],
+            ),
+            (
+                "\n\t",
+                vec![
+                    SerializableToken::LineBreak,
+                    SerializableToken::Text("\t".to_string()), // TODO: This should be updated upstream to handle tabs as whitespace
+                ],
+            ),
+            (
+                "text\ttext",
+                vec![SerializableToken::Text("text\ttext".to_string())], // TODO: This should be updated upstream to handle tabs as whitespace
+            ),
+        ];
+
+        for (input, expected) in test_cases {
+            let tokens = parse(input);
+            assert_eq!(tokens, expected, "Failed for input: {:?}", input);
+        }
+    }
+
+    #[test]
+    fn test_text_edge_cases() {
+        let test_cases = vec![
+            (
+                "Hello, 世界!",
+                vec![SerializableToken::Text("Hello, 世界!".to_string())],
+            ),
+            (
+                "Text with emoji 😊",
+                vec![SerializableToken::Text("Text with emoji 😊".to_string())],
+            ),
+            (
+                "Text with \"quotes\"",
+                vec![SerializableToken::Text("Text with \"quotes\"".to_string())],
+            ),
+            (
+                "Text with \\escaped\\ chars",
+                vec![SerializableToken::Text(
+                    "Text with \\escaped\\ chars".to_string(),
+                )],
+            ),
+        ];
+
+        for (input, expected) in test_cases {
+            let tokens = parse(input);
+            assert_eq!(tokens, expected, "Failed for input: {}", input);
+        }
+    }
+
+    #[test]
+    fn test_error_cases() {
+        // Test with a very long string
+        let long_string = "a".repeat(10000);
+        let tokens = parse(&long_string);
+        assert!(!tokens.is_empty(), "Should handle long strings");
+
+        // Test with a string containing null bytes
+        let null_string = "text\0text";
+        let tokens = parse(null_string);
+        assert_eq!(
+            tokens,
+            vec![SerializableToken::Text("text\0text".to_string())],
+            "Should handle null bytes"
+        );
+
+        // Test with invalid UTF-8 (this will panic if not handled properly)
+        let invalid_utf8 = unsafe { String::from_utf8_unchecked(vec![0xFF, 0xFF]) };
```

Comment:
The `#[forbid(unsafe_code)]` no longer allow to use `unsafe` blocks, so you can convert this to `let invalid_utf8 = std::str::from_utf8(&[0xFF, 0xFF]).unwrap();`